In [1]:
import thermo
from thermo.chemical import Chemical

In [2]:
import math
import fluids
from fluids.units import *


Given:  A header with 170 psia saturated steam is feeding a pulp stock digester through 30 feet of 2" Schedule 40 pipe which includes one standard 90 degree elbow and a fully open conventional plug type disc globe valve.  The initial pressure in the digester is atmospheric.

Find:  The proper size check valve an dthe pressure drop.  The valve should be sied so that the disc is fully lifted at the specified flow.

In [3]:
d_2inch = nearest_pipe(NPS=2*u.inch, schedule='40')[1].to(u.inch) #inner pipe dimension

ft = 0.019 #Crane 410, A-27
Vg = 2.6746 #specific volume, feet**3/lb

P1 = 170*u.psi

d_2inch, ft, Vg

(<Quantity(2.0661417322834645, 'inch')>, 0.019, 2.6746)

# Modified Darcy Formula

Find Ks

In [4]:
#Entrance from header, Crane 410, A-30
K_entrance = 0.5

#Exit to digester, Crane 410, A-30
K_exit = 1.0

#pipe
L = 30*u.feet
D = d_2inch.to(u.feet)
K_pipe = ft*L/D

#globe valve
K_globe = 340*ft  #Crane 410, A-30

#90 degree elbow
K_elbow = 30*ft

Ktot = sum([K_pipe, K_globe, K_elbow, K_entrance, K_exit])
Ktot

<Quantity(11.840518292682926, 'dimensionless')>

DeltaP/P1

In [5]:
#DeltaP/P1, the change in pressure divide by entrance pressure
deltaP_P1 = (170 - 14.7)/170

deltaP_P1

0.9135294117647059

Using the chart on page A-23, Crane 410, for k=1.3, it is found that for K=11.84, the maximum deltaP/P1 is 0.785 (interpolated from table on page A-23).  Since deltaP/P1 is less than indicated by above calculation of deltaP/P1=0.913, sonic velocity occurs at the end of the pipe, and we need to use a different deltaP in the Darcy Formula.
Think of this as the maximum deltaP that can have an effect on the flow through the pipe.

In [7]:
deltaP_P1_max_possible = 0.785

deltaP = deltaP_P1_max_possible*170

deltaP

Y = 0.710 #Crane 410, A-23 interpolated from table

W = (1890*Y*(d_2inch**2)*math.sqrt(deltaP/(Ktot*Vg))).magnitude*u.lb/u.hour

'mass flow rate {:.2f}'.format(W)

'mass flow rate 11759.39 pound / hour'